onebusaway-android/src/androidTest/java/org/onebusaway/android/util/test/UIUtilTest.java

```diff
@@ -552,6 +553,75 @@ public void testArrivalInfoLabels() {
         assertEquals("On time", arrivalInfo.get(29).getStatusText());
         assertEquals("9 min delay", arrivalInfo.get(30).getStatusText());
         assertEquals("On time", arrivalInfo.get(31).getStatusText());
+
+        /**
+         * Test notification texts
+         */
+
+        assertEquals("Route " + arrivalInfo.get(0).getInfo().getShortName()
+                + " has arrived.", arrivalInfo.get(0).getNotifyText());
+        assertEquals("Route " + arrivalInfo.get(1).getInfo().getShortName()
+                + " has departed.", arrivalInfo.get(1).getNotifyText());
+        assertEquals("Route " + arrivalInfo.get(2).getInfo().getShortName()
+                + " has departed.", arrivalInfo.get(2).getNotifyText());
+        assertEquals("Route " + arrivalInfo.get(3).getInfo().getShortName()
+                + " has arrived.", arrivalInfo.get(3).getNotifyText());
+        assertEquals("Route " + arrivalInfo.get(4).getInfo().getShortName()
+                + " has departed.", arrivalInfo.get(4).getNotifyText());
+        assertEquals("Route " + arrivalInfo.get(5).getInfo().getShortName()
+                + " has departed.", arrivalInfo.get(5).getNotifyText());
```

Comment:
Related to my comment in `computeNotifyText()` - I don't think there are currently any unit tests that cover the "NOW" case of an arrival and departure - we should add some to make sure all the IF branches are covered - we can use another raw mocked response if needed.